### PR TITLE
Drop vbox and use qemu everywhere with EFI

### PR DESCRIPTION
**Pull request by Itxaka**
Signed-off-by: itxaka <igarcia@suse.com>

### CODE REVIEW

**Review by kkaempf — CHANGES_REQUESTED** [review]

File: .github/workflows/build-main-teal-x86_64.yaml [hunk 1]
```diff
@@ -518,12 +452,34 @@ jobs:
       - name: Download vagrant box
```
[the remark this answers is not in the document]
Comment (Itxaka) [reply .github/workflows/build-main-teal-x86_64.yaml:452]: It is, we just dont generate the vbox one, and instead move to use qemu everywhere with efi

[the remark this answers is not in the document]
Comment (kkaempf) [reply .github/workflows/build-main-teal-x86_64.yaml:452]: So what's the correct `name:` then ?

[the remark this answers is not in the document]
Comment (Itxaka) [reply .github/workflows/build-main-teal-x86_64.yaml:452]: `Download vagrant box` :P

Both are vagrant boxes, they are independent of the driver used. You can see that the artifact name has changed just below on line 455 ;)

**Review by kkaempf — APPROVED** [review]
We should have documentation about the test infrastructure ...

**Comment by Itxaka** [comment]
> We should have documentation about the test infrastructure ...

We have some under the development.md !

**Comment by Itxaka** [comment]
for squashfs images a mod is missing somehow:

`error: ../../grub-core/fs/fshelp.c:258:file /grub2/arm64-efi/xzio.mod not found`

**Pull request merged**